package lf;

message LFunction {
  required uint32 degree         = 1;
  repeated uint32 signature      = 2 [packed = true];
  repeated double real_shiftsR   = 3 [packed = true];
  repeated double real_shiftsC   = 4 [packed = true];
  repeated uint32 character      = 5;
  required uint32 level          = 6;
  required uint32 character_mod  = 7;
  required uint32 character_nr   = 8;
  required uint32 Gamma_factor_precision = 9;

  repeated float imaginary_shiftsR = 10 [packed = true];
  repeated float imaginary_shiftsC = 11 [packed = true];
  required float logsign           = 12;
  required bool  arithmetic        = 13;
  required bool  self_dual         = 14;
  required bool  primitive         = 15;

  message CoeffDecimal {
     required uint32  precision     = 1;
     repeated string  coefficients  = 2;
  }

  required CoeffDecimal coefficients_decimal = 16;
}
